src/TokenClient.ts

```diff
@@ -112,7 +112,7 @@ export class TokenClient {
         let basicAuth: string | undefined;
         switch (this._settings.client_authentication) {
             case "client_secret_basic":
-                if (!client_secret) {
+                if (client_secret === undefined || client_secret === null) {
```

Comment:
`!client_secret` is the same as `client_secret === undefined || client_secret === null`, both evaluate to true when  `client_secret` is null or undefined